src/runtime/execution.rs

```diff
@@ -304,14 +322,23 @@ impl ExecutionState {
     {
         Self::with(|state| {
             let task_id = TaskId(state.tasks.len());
+            let is_some = state.try_current().is_some();
+
             let clock = if let Some(ref mut clock) = initial_clock {
                 clock
             } else {
                 // Inherit the clock of the parent thread (which spawned this task)
                 state.increment_clock_mut()
             };
             clock.extend(task_id); // and extend it with an entry for the new thread
-            let task = Task::from_closure(f, stack_size, task_id, name, clock.clone());
+            let clock = clock.clone();
+
+            let current_span = tracing::Span::current();
+            let span = if is_some { &state.current().span } else { &current_span };
```

Comment:
Once the reference isn't required for `from_closure` I'd just inline `is_some` here.